infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerimage/EntryPointParser.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.dockerimage;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.StreamSupport;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalMachineConfig;
+
+/** Can be used to parse container entry-point definition specified as a YAML list. */
+public final class EntryPointParser {
+  private final YAMLMapper mapper = new YAMLMapper();
+
+  /**
+   * Parses the attributes contained in the provided machine config and produces an entry point
+   * definition.
+   *
+   * @param machineConfig the machine config to extract the entry point info from
+   * @return an entry point definition, never null
+   * @throws InfrastructureException on failure to parse the command or arguments
+   */
+  public EntryPoint parse(InternalMachineConfig machineConfig) throws InfrastructureException {
+
+    if (machineConfig == null) {
+      return new EntryPoint(emptyList(), emptyList());
+    }
+
+    String command = machineConfig.getAttributes().get(MachineConfig.CONTAINER_COMMAND_ATTRIBUTE);
+    String args = machineConfig.getAttributes().get(MachineConfig.CONTAINER_ARGS_ATTRIBUTE);
+
+    if (command == null) {
+      return new EntryPoint(emptyList(), emptyList());
+    }
+
+    try {
+      List<String> commandList = parseAsList(command);
+      List<String> argList = args == null ? emptyList() : parseAsList(args);
+
+      return new EntryPoint(commandList, argList);
+    } catch (IOException e) {
+      throw new InfrastructureException(
+          "Failed to apply entry point configuration for container.", e);
```

Comment:
Looks like a different message should be here

Reply:
Huh, yes, remnant of the original location of the code :man_facepalming: